easybuild/easyblocks/o/openssl_wrapper.py

```diff
@@ -0,0 +1,269 @@
+"""
+import os
+import re
+
+from easybuild.easyblocks.generic.bundle import Bundle
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.filetools import change_dir, expand_glob_paths, mkdir, read_file, symlink, which
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import DARWIN, LINUX, get_os_type, get_shared_lib_ext, find_library_path
+from easybuild.tools.build_log import EasyBuildError, print_warning
+
+
+class EB_OpenSSL_wrapper(Bundle):
+    """
+    Locate the installation files of OpenSSL in the host system
+    If available, wrap the system OpenSSL by symlinking all installation files
+    Fall back to the bundled component otherwise
+    """
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Easyconfig parameters specific to OpenSSL wrapper"""
+        extra_vars = Bundle.extra_options(extra_vars=extra_vars)
+        extra_vars.update({
+            'wrap_system_openssl': [True, 'Detect and wrap OpenSSL installation in host system', CUSTOM],
+        })
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Locate the installation files of OpenSSL in the host system"""
+        super(EB_OpenSSL_wrapper, self).__init__(*args, **kwargs)
+
+        # Libraries packaged in OpenSSL
+        openssl_libs = ['libssl', 'libcrypto']
+        # list of relevant library extensions per system and version of OpenSSL
+        # the first item should be the extension of an installation from source,
+        # it will be used in the sanity checks of the component
+        openssl_libext = {
+            '1.0': {
+                LINUX: ('so.1.0.0', 'so.10'),
+                DARWIN: ('1.0.dylib', ),
+            },
+            '1.1': {
+                LINUX: ('so.1.1', ),
+                DARWIN: ('1.1.dylib', ),
+            },
+        }
+
+        os_type = get_os_type()
+        if self.version in openssl_libext and os_type in openssl_libext[self.version]:
+            # generate matrix of versioned .so filenames
+            self.openssl_libs = [
+                ['%s.%s' % (lib, ext) for ext in openssl_libext[self.version][os_type]]
+                for lib in openssl_libs
+            ]
+        else:
+            raise EasyBuildError("Don't know name of OpenSSL system library for version %s and OS type %s",
+                                 self.version, os_type)
+
+        # folders containing engines libraries
+        self.openssl_engines = {
+            '1.0': 'engines',
+            '1.1': 'engines-1.1',
+        }
+
+        # Paths to system libraries and headers of OpenSSL
+        self.ssl_syslib = None
+        self.ssl_sysheader = None
+        self.ssl_sysengines = None
+        self.ssl_sysbin = None
+
+        if not self.cfg.get('wrap_system_openssl'):
+            return
+
+        # Check the system libraries of OpenSSL
+        for n, libssl in enumerate(self.openssl_libs[0]):
```

Comment:
Don't use single-letter variables outside of list comprehensions please. I'd go with `idx` here.

Reply:
fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2429/commits/724ea6ec2fe2bc2b99e0c74a56820056056edf90